### PR TITLE
fix(DataStore): Cascade delete sync from children to parent models

### PR DESCRIPTION
**Pull request by lawmicha**
*Issue #, if available:*

*Description of changes:*
Change the logic to align with JS's logic to "syncDeletions from children to parent" from previously "sync deletions from parent to children". And added unit testing around CascadeDeleteOperation 

*Check points: (check or cross out if not relevant)*

- [x] Added new tests to cover change, if needed
- [x] Build succeeds with all target using Swift Package Manager
- [x] All unit tests pass
- [ ] All integration tests pass
- [x] Security oriented best practices and standards are followed (e.g. using input sanitization, principle of least privilege, etc)
- [ ] Documentation update for the change if required
- [x] PR title conforms to conventional commit style
- [ ] ~If breaking change, documentation/changelog update with migration instructions~

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginTests/Storage/CascadeDeleteOperationTests.swift [hunk 1]
```diff
@@ -0,0 +1,656 @@
...
+            }
+        }
+        operation.start()
+        wait(for: [completed], timeout: 1)
+        guard case .success(let queriedRestaurants) = queryModelSynchronous(modelType: Restaurant.self,
+                                                                            predicate: predicate) else {
+            XCTFail("Failed to query")
+            return
+        }
+        XCTAssertEqual(queriedRestaurants.count, 1)
```
Comment (thisisabhash) [inline AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginTests/Storage/CascadeDeleteOperationTests.swift:174]: nit: can add an assert for name equality.

File: AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginTests/Storage/CascadeDeleteOperationTests.swift [hunk 2]
```diff
@@ -0,0 +1,656 @@
...
```
Comment (thisisabhash) [inline AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginTests/Storage/CascadeDeleteOperationTests.swift:255]: nit: can add assert for name matching

**Review by thisisabhash — APPROVED** [review]
LGTM

**Comment by lawmicha** [comment]
Will merge this and https://github.com/aws-amplify/amplify-ios/pull/1730 together to `main` in this PR, and close out https://github.com/aws-amplify/amplify-ios/pull/1730 which has been reviewed

**Pull request merged**